src/api/auth.ts

```diff
@@ -346,13 +359,32 @@ export function changeUserRole(
 
 export function publishPackage(
   pkgName: string, username: string, organization: string,
-  version: string, auth: any): Promise<string> {
-    return new Promise((resolve, reject) => {
-      if (!organization) {
+  teamPermissions: [TeamPermissionData], version: string, auth: any): Promise<string> {
+    let pkgObject: any = null;
+    getPackage(pkgName, auth).then(p => pkgObject = p);
```

Comment:
you can start here with `return getPackage(...` and you never need to use `return new Promise` after. 
we can chat on Slack about this one if you need help.